src/Tests/dotnet-add-package.Tests/GivenDotnetPackageAdd.cs

```diff
@@ -41,6 +41,60 @@ public void WhenValidPackageIsPassedBeforeVersionItGetsAdded()
             cmd.StdErr.Should().BeEmpty();
         }
 
+        public static readonly List<object[]> AddPkg_PackageVersionsLatestPrereleaseSucessData
+            = new List<object[]>
+            {
+                    new object[] { new string[] { "0.0.5", "0.9.0", "1.0.0-preview.3" }, "1.0.0-preview.3" },
+                    new object[] { new string[] { "0.0.5", "0.9.0", "1.0.0-preview.3", "1.1.1-preview.7" }, "1.1.1-preview.7" },
+                    new object[] { new string[] { "0.0.5", "0.9.0", "1.0.0" }, "1.0.0" },
+                    new object[] { new string[] { "0.0.5", "0.9.0", "1.0.0-preview.3", "2.0.0" }, "2.0.0" },
+                    new object[] { new string[] { "1.0.0-preview.1", "1.0.0-preview.2", "1.0.0-preview.3" }, "1.0.0-preview.3" },
+            };
+
+        [Theory]
+        [MemberData(nameof(AddPkg_PackageVersionsLatestPrereleaseSucessData))]
+        public void WhenPrereleaseOptionIsPassed(string[] inputVersions, string expectedVersion)
+        {
+            var targetFramework = "net5.0";
+            TestProject testProject = new TestProject()
+            {
+                Name = "Project",
+                IsSdkProject = true,
+                IsExe = false,
+                TargetFrameworks = targetFramework,
+            };
+
+            var packages = inputVersions.Select(e => GetPackagePath(targetFramework, "A", e)).ToArray(); 
+
+            testProject.AdditionalProperties.Add("RestoreSources",
```

Comment:
Curious, this property is how to mock a feed?

As in you can simulate a restore with package with only given versions?

Reply:
The sources specified through an msbuild property take preferrence over nuget.config and all. 

This was my recommendation because it allowed fully locked down feeds.